internal/provider/common/role.go

```diff
@@ -16,6 +16,11 @@ import (
 	"github.com/samber/lo"
 )
 
+type Role struct {
+	Role string
+	Id   string
+}
```

Comment:
we should also move this to a directory just for tests
don't want it to accidentally be used in the business logic since Role and EntityId could be used incorrectly if we don't know the entityType